packages/design-system/src/components/Button/Button.test.tsx

```diff
@@ -90,6 +90,19 @@ describe('Button', () => {
     expect(inputRef.current.tagName).toEqual('BUTTON');
   });
 
+  it('forwards a mutable object inputRef', () => {
+    const MyComponent = () => {
+      const inputRef = useRef<HTMLButtonElement>();
+      useEffect(() => {
```

Comment:
I'm curious about the usage of `useEffect` in tests, as I haven't encountered it much before. Would using `getByRole` to locate the element and then asserting the `ref.current` value work in this case? Or is there a specific reason we need `useEffect` here instead?

Reply:
I didn't think I could access that value outside my test component, so that's why I put my assertion in a useEffect. I guess I could `let inputRef` define it outside of `MyComponent`, but I didn't know if that would produce unexpected results. Seemed the most straightforward to put the assertion in the normal render cycle of `MyComponent`

Reply:
Ok, that make sense!